jtransc-rt/resources/d/Base.d

```diff
@@ -236,8 +236,28 @@ class N {
 
 	static public long i2j(int v) { return v; }
 
-	static public long f2j(float v) { return cast(long)(v); }
-	static public long d2j(double v) { return cast(long)(v); }
+	static public long f2j(float v) {
+		if (std.math.isNaN(v)) {
```

Comment:
I had the same issue. Some languages doesn't have it. Just have isInfinity, which is false for nan